content/posts/lets-code-contact-form-in-phoenix-part-one.md

```diff
@@ -31,17 +31,27 @@ end
 
 Here, I define a `Support.Message` module to model the user's message. Since this is an internal module to the `Support` context, I specify `@moduledoc false`. [This convention](https://hexdocs.pm/elixir/writing-documentation.html#hiding-internal-modules-and-functions) hides the module from our application's documentation. I typically document modules that are part of the public interface of my application and hide the rest.
 
-The `@type t` convention documents the type this module models. In this case, it's a struct with an `email`, `subject`, and `body` attributes. Simple enough so far. Note, this module will not compile because a struct has not been defined. Next, I will add a schema using [the Ecto library](https://hexdocs.pm/ecto/Ecto.html) which will give us the struct and allow this module to be compiled.
+The `@type t` convention documents the type this module models. In this case, it's a struct of this module, denoted with the [`__MODULE__` convenience function](https://hexdocs.pm/elixir/Kernel.SpecialForms.html#__MODULE__/0), and containing `email`, `subject`, and `body` attributes. Simple enough so far. Note, this module will not compile because a struct has not been defined. Next, I will add a schema using [the Ecto library](https://hexdocs.pm/ecto/Ecto.html) which will give us the struct and allow this module to be compiled.
 
 ## Using Ecto's Embedded Schemas
 
-Since Ecto ships with most Phoenix applications, I use it here to structure my new `Message`. As mentioned before, Elixir structs are a perfectly viable option. But Ecto provides us with changesets and validations, which I will leverage later. Before that, I need to define the schema.
+Since Ecto ships with most Phoenix applications, I use it here to structure my
+new `Message`. As mentioned before, Elixir structs are a perfectly viable
```

Comment:
This may be a bit too big of a suggestion, but I can almost see dropping "as mentioned" and beyond and joining this first sentence with the next paragraph. For some reason I really wanted you to tag up with the idea that you are doing this even though it's not a database-backed structure.

I'd be interested to see how it reads if, after the quote, you call out that you could use structs but gain all the things mentioned (basically the end of this paragraph). I think that gets me sold on Ecto, but lets me know _you know_ you don't have to use Ecto.